### ISSUE TITLE
Output sourcemaps in consistent format

### ISSUE THREAD

**Issue by jenius** [jenius/accord#60]
So right now, every compiler returns the sourcemap however it pleases. While this is true to how the compilers work and less intrusive on accord's part, I'm of the opinion that this is not actually a benefit. My theory is that anyone using accord is probably compiling from a number of different languages, which is why they are using accord, for consistency between them. But if the sourcemaps are all returned in random formats, this makes everything less consistent.

So perhaps we could use [this library](https://github.com/thlorenz/convert-source-map) to parse out the resulting sourcemaps and ensure that they are all returned as JSON with consistent formatting, to make things more consistent across the board.

**Comment by slang800** [comment]
Already implemented this in https://github.com/slang800/accord/blob/master/lib/job.coffee

**Comment by jenius** [comment]
You keep mentioning these things you have implemented, but I don't have a PR still for this, and we're trying to integrate these changes into roots and other libraries right now. Is there going to be a PR ready soon? If not, I'm just going to implement this now, and you can remove it when you do have the PR ready.

**Comment by jenius** [comment]
Also @slang800 have you implemented these changes individually in clean branches, or globbed them together with massive architectural and stylistic refactors that might not be approved? The former would obviously be much better but I'm a little afraid that's not the case based on my quick review of the code linked above.

**Comment by slang800** [comment]
It's waiting on #55 and is in #47

**Comment by slang800** [comment]
And it's on top of architectural changes because they're necessary for a clean implementation.

**Comment by jenius** [comment]
Oh boy. Breaking down #47 a bit would still help quite a lot. But I'll take a crack at these. Typically you want to start small with changes like this rather than just going ahead and rewriting everything. It's really hard to get PRs merged into libraries when you rewrite everything, you have to take it step by step.

I'll take a closer look at these this week. Are they finished in your opinion?

**Comment by slang800** [comment]
Yeah, #47 is months of changes... at one point it was a reasonable-sized PR, but it sat in review-purgatory and grew over time. #55 should be ready, and is the next chunk of #47 to go in.

#47 as a whole should also be ready (as in it works and passes all the tests), but it'll probably need to be broken up for review.

**Comment by jenius** [comment]
Ok so at the moment neither of these PRs are ready. I'm going to continue working on the code we have at the moment in the interest of moving forward, but will also make sure to review these PRs as they are cleaned up and split out the way they should be.

**Comment by slang800** [comment]
#55 should be ready now, and I'll bring in the next chunk as soon as that's merged
